### PR TITLE
Saf 156 payment validation

### PR DESCRIPTION
**Pull request by Alfr3d**

### CODE REVIEW

File: controllers/admin/AdminSaferPayOfficialPaymentController.php [hunk 1]
```diff
@@ -69,8 +70,13 @@ public function postProcess()
         /** @var SaferPayRestrictionCreator $restrictionCreator */
         $restrictionCreator = $this->module->getModuleContainer()->get(SaferPayRestrictionCreator::class);
 
+        $paymentMethodsFromSaferPay = $this->getPaymentMethods();
```
Comment (tomjas1997) [inline controllers/admin/AdminSaferPayOfficialPaymentController.php:73]: Wouldn't use "SaferPay" in the variable name, only paymentMethods should be enough.

File: controllers/admin/AdminSaferPayOfficialPaymentController.php [hunk 2]
```diff
@@ -215,7 +226,28 @@ protected function initForm()
             'name' => 'all',
             'form_group_class' => 'saferpay-group all-payments',
         ];
-        foreach (SaferPayConfig::PAYMENT_METHODS as $paymentMethod) {
+
+        try {
+            /** @var \Invertus\SaferPay\Service\SaferPayObtainPaymentMethods $saferPayObtainPaymentMethods */
+            $saferPayObtainPaymentMethods = $this->module->getModuleContainer()->get(SaferPayObtainPaymentMethods::class);
+            $paymentMethodsFromSaferPay = $saferPayObtainPaymentMethods->obtainPaymentMethodsNamesAsArray();
```
Comment (tomjas1997) [inline controllers/admin/AdminSaferPayOfficialPaymentController.php:233]: same here

File: controllers/admin/AdminSaferPayOfficialPaymentController.php [hunk 3]
```diff
@@ -235,4 +267,28 @@ protected function initForm()
...
+            $saferPayErrors = json_decode($this->context->cookie->saferPayErrors, true);
+            $saferPayErrors[] = $exceptionService->getErrorMessageForException(
+                $exception,
+                $exceptionService->getErrorMessages()
+            );
+            $this->context->cookie->saferPayErrors = json_encode($saferPayErrors);
+
+            $this->errors[] = $this->l('To see available payment methods, you must connect to your SaferPay account.');
+
+            return null;
```
Comment (tomjas1997) [inline controllers/admin/AdminSaferPayOfficialPaymentController.php:291]: Maybe should return empty array and then there would be no need to check if it is null, foreach'ing empty array would display nothing

File: src/Api/ApiRequest.php [hunk 4]
```diff
@@ -86,7 +86,7 @@ public function get($url, $params = [])
         } catch (\Exception $exception) {
             $logs = new SaferPayLog();
             $logs->message = $exception->getResponse()->getBody()->getContents();
-            $logs->payload = $params['body'];
+            $logs->payload = $exception->getCode();
```
Comment (tomjas1997) [inline src/Api/ApiRequest.php:89]: Pretty sure payload shouldn't just be status code :D

File: src/Install/Installer.php [hunk 5]
```diff
@@ -78,96 +78,9 @@ public function install()
...
-        $logoCreator = $this->module->getModuleContainer()->get(SaferPayLogoCreator::class);
-
-        /** @var SaferPayRestrictionCreator $restrictionCreator */
-        $restrictionCreator = $this->module->getModuleContainer()->get(SaferPayRestrictionCreator::class);
-
-        $success = true;
-
-        foreach (SaferPayConfig::PAYMENT_METHODS as $paymentMethod) {
-            $success &= $this->enablePayment($paymentMethod, $paymentCreator);
-            $success &= $this->enableLogo($paymentMethod, $logoCreator);
```
Comment (tomjas1997) [inline src/Install/Installer.php:107]: You are checking if logo is enabled for displaying payment options but wouldn't deletion of this installer make that everything was disabled on installation?

Comment (Alfr3d) [reply src/Install/Installer.php:107]: Yes, in this case merchant must connect to saferpay account first and only after that he will be able enable payments 😅

**Pull request merged**